documentation/projects/proposals/data_normalization/20240227-implementation_plan_catalog_data_cleaning.md

```diff
@@ -0,0 +1,198 @@
+[Expected Outcomes](#expected-outcomes) section below.
+
+## Overview
+
+This document describes a solution for incorrect data in the catalog database
+(DB) that has to be cleaned up every time a data refresh is run, avoiding wasted
+resources.
+
+## Background
+
+One of the steps of the [data refresh process for images][img-data-refresh] is
+cleaning the data that is not fit for production. This process is triggered
+weekly by an Airflow DAG, and then runs in the Ingestion Server, taking
+approximately just over **20 hours** to complete, according to a inspection of
+latest executions. The cleaned data is only saved to the API database, which is
+replaced each time during the same data refresh, causing it to have to be
+repeated each time to make the _same_ corrections.
+
+This cleaning process was designed this way to speed the rows update up since
+the relevant part was to provide the correct data to users via the API. Most of
+the rows affected were added previous to the creation of the `MediaStore` class
+in the Catalog (possibly by the discontinued CommonCrawl ingestion) which is
+nowadays responsible for validating the provider data. However, it entails a
+problem of wasting resources both in time, which continues to increase, and in
+the machines (CPU) it uses, which could easily be avoided making the changes
+permanent by saving them in the upstream database.
+
+[img-data-refresh]:
+  https://github.com/WordPress/openverse-catalog/blob/main/DAGs.md#image_data_refresh
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+- The catalog database (upstream) preserves the cleaned data results of the
+  current Ingestion Server's cleaning steps
+- The image Data Refresh process is simplified by reducing the cleaning steps
+  time to nearly zero (and optionally removing them).
+
+## Step-by-step plan
+
+The cleaning functions that the Ingestion Server applies are already implemented
+in the Catalog in the `MediaStore` class: see its `_tag_blacklisted` method
+(which probably should be renamed) and the [url utilities][url_utils] file. The
+only part that it's not there and can't be ported is the filtering of
+low-confidence tags, since provider scripts don't save an "accuracy" by tag.
+
+With this the plan then starts in the Ingestion Server with the following steps:
+
+1. [Save TSV files of cleaned data to AWS S3](#save-tsv-files-of-cleaned-data-to-aws-s3)
+1. [Make and run a batched update DAG for one-time cleanup](#make-and-run-a-batched-update-dag-for-one-time-cleanup)
+1. [Run an image Data Refresh to confirm cleaning time is reduced](#run-an-image-data-refresh-to-confirm-cleaning-time-is-reduced)
+
+[url_utils]:
+  https://github.com/WordPress/openverse/blob/a930ee0f1f116bac77cf56d1fb0923989613df6d/catalog/dags/common/urls.py
+
+## Step details
+
+### Save TSV files of cleaned data to AWS S3
+
+In a previous exploration, it was set to store TSV files of the cleaned data in
```

Comment:
```suggestion
In a previous exploration, the Ingestion Server was set to store TSV files of the cleaned data in
```

Do you mind also linking the PR where we added this for context?

Reply:
Done ✅